src/http/h1/encode.rs

```diff
@@ -107,50 +108,54 @@ impl Chunked {
             Chunked::End => return Ok(0),
             _ => {}
         }
+
         let mut n = {
-            let pieces = match *self {
+            // TODO: Find a better way to concatente
+            //    Could use write! if we reserved the required capacity upfront
+            match *self {
```

Comment:
The compiler can inline itself. I've left the way it is as a reminder to investigate using `AsyncWrite::write_bufs` in the future, but I know that it's only a reminder to me, no one else would be able to determine that. Probably could use a comment to that effect. If you'd rather inline the implementation and have a comment there instead, that's fine.